service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java

```diff
@@ -193,13 +201,17 @@ private void processSamples(List<Sample> samples, String projection) {
             
             Set<Integer> samplesWithCopyNumberSegMap = new HashSet<>();
             samplesWithCopyNumberSegMap.addAll(samplesWithCopyNumberSeg);
-           
+
             samples.forEach(sample -> {
                 sample.setSequenced(sequencedSampleIdsMap.get(sample.getCancerStudyIdentifier())
                     .contains(sample.getStableId()));
                 sample.setCopyNumberSegmentPresent(samplesWithCopyNumberSegMap.contains(sample.getInternalId()));
-                if (!structuralVariantSampleIdsMap.isEmpty()) {
-                    sample.setProfiledForFusions(structuralVariantSampleIdsMap.get(sample.getCancerStudyIdentifier()).contains(sample.getStableId()));
+                if (studiesProfiledWithSVs.contains(sample.getCancerStudyIdentifier())) {
+                    if (!structuralVariantSampleIdsMap.get(sample.getCancerStudyIdentifier()).isEmpty()) {
+                        sample.setProfiledForFusions(structuralVariantSampleIdsMap.get(sample.getCancerStudyIdentifier()).contains(sample.getStableId()));
+                    } else {
+                        sample.setProfiledForFusions(sequencedSampleIdsMap.get(sample.getCancerStudyIdentifier()).contains(sample.getStableId()));
+                    }
```

Comment:
might be good to have a comment here. Something like:

```
/*
TODO: Eventually all studies with STRUCTURAL_VARIANT data should have case lists, so there should always be an entry in `structuralVariantSampleIdsMap`. This case is to support old `FUSION` data in the mutations table that don't have case lists. In that case we assume any sample that has been sequenced to have been profiled for fusions as well
*/
```
Makes sense?

Reply:
It would be good to create an issue as well pointing to these various parts in the code where FUSION data should be cleaned up. I think by having `FUSION` specifically as a comment it will be easy to find for whoever tries to remove all `FUSION` related parts from the code.

Reply:
good idea, @inodb!